Java/LFU.java

```diff
@@ -0,0 +1,144 @@
+package lfu;
+import java.util.LinkedList;
+import java.util.ListIterator;
+import java.util.HashMap;
+
+/**
+ * @author salman
+ * @param <T>
+ */
+
+public class LFU<T> {
+
+	private int MAX_SIZE = 2;
+
+	private LinkedList< FrequencyList<T> > lfuList;
+	private HashMap<T, Pair<Integer, ListIterator<T> > > nodeFrequencyMap;
```

Comment:
You can only iterate through an interator once. But if you take an iterable, you can easily call .iterator() on it, to get it's corresponding iterator.

Plus, ListIterator is too restrictive. I don't think you need any aspect of ListIterator that is not available in Iterator itself.